src/index.ts

```diff
@@ -57,18 +61,42 @@ function applyNonce(req: Request, cspString: string): string {
 
 function applyAutoTld(req: Request, cspString: string, domainOptions?: ParseOptions): string {
     if (cspString.includes(TLD)) {
-        let domain = parseDomain(req.hostname, domainOptions);
+        let tld = parseDomain(req.hostname, domainOptions);
 
-        if (!domain || !domain.tld) {
+        if (!tld) {
             return cspString;
         }
 
-        return cspString.replace(new RegExp(TLD, 'g'), domain.tld);
+        return cspString.replace(new RegExp(TLD, 'g'), tld);
     }
 
     return cspString;
 }
 
+function parseDomain(hostname: string, domainOptions?: ParseOptions): string | null {
+	let customTlds = domainOptions?.customTlds;
+	if (customTlds instanceof RegExp) {
+		const tld = hostname.match(customTlds);
+		if (tld !== null) {
+			return tld[0].replace(/^\.+/, '');
+		}
+	} else if (customTlds) {
```

Comment:
```suggestion
	}

	if (Array.isArray(customTlds)) {
```

Reply:
Hah! The code looked like that to start with, but then I changed it to this (to be more similar to what was there before). Fixed.